context/src/main/java/io/opentelemetry/context/ContextKey.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.context
+
+/**
+ * Key for indexing values of type [T] stored in a [Context]. [ContextKey] are
+ * compared by reference, so it is expected that only one [ContextKey] is created for a
+ * particular type of context value.
+ *
+ * <pre>`public class ContextUser {
+ *
+ * private static final ContextKey<MyState> KEY = ContextKey.named("MyState");
+ *
+ * public Context startWork() {
+ * return Context.withValues(KEY, new MyState());
+ * }
+ *
+ * public void continueWork(Context context) {
+ * MyState state = context.get(KEY);
+ * // Keys are compared by reference only.
+ * assert state != Context.current().get(ContextKey.named("MyState"));
+ * ...
+ * }
+ * }
+ *
+`</pre> *
+ */
+// ErrorProne false positive, this is used for its type constraint, not only as a bag of statics.
+interface ContextKey<T> {
```

Comment:
Requirement of spec that the type is 'opaque' so we likely don't want to return a string here https://opentelemetry.io/docs/specs/otel/context/#create-a-key